src/runtime_src/core/pcie/windows/device_windows.cpp

```diff
@@ -835,6 +835,83 @@ struct rom
   }
 };
 
+struct data_retention
+{
+  using result_type = uint32_t;
+  using value_type = uint32_t;
+  
+  static result_type
+  user_get(const xrt_core::device* device)
+  {
+	  return 0;
+  }
+
+  static result_type
+  mgmt_get(const xrt_core::device* device)
+  {
+    auto init_ret = [](const xrt_core::device* dev) {
+    uint32_t ret;
+    mgmtpf::get_data_retention(dev->get_mgmt_handle(), &ret);
+    return ret;
+    };
+
+    static std::map<const xrt_core::device*, uint32_t> info_map;
+    static std::mutex mutex;
+    std::lock_guard<std::mutex> lk(mutex);
+    auto it = info_map.find(device);
+    if (it == info_map.end()) {
+      auto ret = info_map.emplace(device,init_ret(device));
+      it = ret.first;
+    }
+
+    return (*it).second;
+  }
+
+  static void
+  user_put(const xrt_core::device* device, value_type)
+  {
+    throw std::runtime_error("device data retention query is not implemented on user windows");
```

Comment:
Is this missing functionality.  I as for the check-in comments "suggests" this functionality is being added.  

If this truly the intent, please add a comment why it isn't implemented (e.g., missing driver support)